taf/tools/repo/__init__.py

```diff
@@ -133,30 +143,90 @@ def update(path, url, clients_library_dir, from_fs, expected_repo_type,
         Update can be in strict or no-strict mode. Strict mode is set by specifying --strict, which will raise errors
         during update if any/all warnings are found. By default, --strict is disabled.
         """
-        if path is None and clients_library_dir is None:
+        if path is None and library_dir is None:
             raise click.UsageError('Must specify either authentication repository path or library directory!')
 
         if profile:
-            import cProfile
-            import atexit
+            start_profiling()
+
+        config = RepositoryConfig(
```

Comment:
Can we get into a situation where we unintentionally misuse the new RepositoryConfig when we forget to specify some field that's expected in a function? Do we need to add validation to all functions that accept RepositoryConfig object that check if all necessary fields are specified (that that function expects to use), before continuing with other code?

Reply:
The check is done in `updater.py`. For example:
```
   if config.url is None:
        raise UpdateFailedError("URL has to be specified when cloning repositories")
```
Default values are also set here. So I think that it should be fine